software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,472 @@
+               'fclass','fmv_x_w','local_ld','local_st',
+               'remote_ld_dram','remote_ld_global','remote_ld_group',
+               'remote_st_dram','remote_st_global','remote_st_group',
+               'local_flw','local_fsw','remote_flw','remote_fsw',
+               'lr','lr_aq','swap_aq','swap_rl','beq','bne',
+               'blt','bge','bltu','bgeu','jalr','jal', 'sll',
+               'slli','srl','srli','sra','srai','add','addi','sub',
+               'lui','auipc','xor','xori','or','ori','and','andi',
+               'slt','slti','sltu','sltiu','mul','mulh','mulhsu',
+               'mulhu','div','divu','rem','remu','fence']
+
+miss_list   = ['icache_miss', 'beq_miss', 'bne_miss', 'blt_miss',
+               'bge_miss', 'bltu_miss', 'bgeu_miss', 'jalr_miss']
+
+stalls_list = ['stall_fp_remote_load','stall_fp_local_load',
+               'stall_depend',
+               'stall_depend_remote_load_dram',
+               'stall_depend_remote_load_global',
+               'stall_depend_remote_load_group',
+               'stall_depend_local_load','stall_force_wb',
+               'stall_ifetch_wait','stall_icache_store',
+               'stall_lr_aq','stall_md','stall_remote_req','stall_local_flw']
+
+
+
+class VanillaStatsParser:
+
+  # default constructor
+  def __init__(self, manycore_dim_y, manycore_dim_x, per_tile_stat):
+
+    self.manycore_dim_y = manycore_dim_y
+    self.manycore_dim_x = manycore_dim_x
+    self.manycore_dim = manycore_dim_y * manycore_dim_x
+    self.per_tile_stat = per_tile_stat
+
+
+    self.max_tile_groups = 1024
+    self.num_tile_groups = 0
+
+    self.timing_start_list = [0] * self.max_tile_groups 
+    self.timing_end_list =   [0] * self.max_tile_groups
+#    self.timing_stat = []
+
+    self.total_execution_time = 0
+    self.total_instr_cnt = 0
+    self.total_stall_cnt = 0
+    self.total_miss_cnt = 0
+
+    self.tile_stat_dict = [[dict() for x in range(self.manycore_dim_x)] for y in range(self.manycore_dim_y)]
+    self.manycore_stat_dict = dict()
+
+
+
+  # private method
+  # parse the stats file to generate total execution time
+  # for each tile group and for total
+  def generate_stats_timing(self, trace):
+    y = trace["y"]
+    x = trace["x"]
+    relative_y = y - bsg_TILE_GROUP_ORG_Y
+    relative_x = x - bsg_TILE_GROUP_ORG_X
+    tg_num = trace["tag"]
+    # Only count those coming from the origin tile (0,0)
+    if (relative_x == 0 and relative_y == 0):
+      if(tg_num < bsg_PRINT_STAT_END_VAL):
+        self.timing_start_list[tg_num] = trace["time"]
+        self.num_tile_groups += 1
+      else:
+        self.timing_end_list[tg_num - bsg_PRINT_STAT_END_VAL] = trace["time"]
+
+
+  # private method
+  # Sum up all other stats for all tiles based on the last bsg_print_stat instr
+  def generate_stats_all(self):
+      # Generate timing stats 
+      for trace in self.traces:
+        self.generate_stats_timing(trace)
+
+      self.timing_stat = [0] * self.num_tile_groups
```

Comment:
Why are you initializing a object member in a non-init method?

Reply:
Because I didn't know the size of the object I needed at init time, but I'll go with the max size possible, it's not like I'm saving that much space anyway. Thanks!

Reply:
You can also use default_dict, or something similar for lists.